sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -674,8 +675,11 @@ def run_ParDo(self, transform_node, options):
     from apache_beam.runners.dataflow.internal import apiclient
     transform_proto = self.proto_context.transforms.get_proto(transform_node)
     transform_id = self.proto_context.transforms.get_id(transform_node)
-    if (apiclient._use_fnapi(options)
-        and transform_proto.spec.urn == common_urns.primitives.PAR_DO.urn):
+    # The data transmitted in SERIALIZED_FN is different depending on whether
+    # this is a fnapi pipeline or not.
+    if (apiclient._use_fnapi(options) and
+        (transform_proto.spec.urn == common_urns.primitives.PAR_DO.urn or
+         apiclient._use_unified_worker(options))):
```

Comment:
Is this to work around issues in the Java harness?

Reply:
The unified worker has all of the transforms handled here use transform ids from the model pipeline as the serialized representation, including e.g. WINDOW_INTO.  For reasons unknown to me, the Java runner harness only uses transform ids for PAR_DO operations but not WINDOW_INTO.  It fails if this change is made when using it.